packages/infoportal-client/src/features/Meal/Pdm/MealPdmCashDashboard.tsx

```diff
@@ -1,69 +1,62 @@
 import {useI18n} from '@/core/i18n'
 import {PeriodPicker} from '@/shared/PeriodPicker/PeriodPicker'
 import {DebouncedInput} from '@/shared/DebouncedInput'
 import {Div, SlidePanel} from '@/shared/PdfLayout/PdfSlide'
 import {DataFilter} from '@/shared/DataFilter/DataFilter'
-import {KoboAnswerFlat, Meal_cashPdm, OblastIndex} from 'infoportal-common'
+import {Meal_cashPdm, OblastIndex} from 'infoportal-common'
 import {DataFilterLayout} from '@/shared/DataFilter/DataFilterLayout'
 import {Page} from '@/shared/Page'
 import {useKoboSchemaContext} from '@/features/KoboSchema/KoboSchemaContext'
-import {useMealPdmContext} from '@/features/Meal/Pdm/MealPdmContext'
+import {PdmData, PdmForm, useMealPdmContext} from '@/features/Meal/Pdm/Context/MealPdmContext'
 import {appConfig} from '@/conf/AppConfig'
 import {ChartBarSingleBy} from '@/shared/charts/ChartBarSingleBy'
 import {ChartPieWidgetBy} from '@/shared/charts/ChartPieWidgetBy'
 import {ChartBarMultipleBy} from '@/shared/charts/ChartBarMultipleBy'
 import {AgeGroupTable} from '@/shared/AgeGroupTable'
 import {Panel, PanelBody} from '@/shared/Panel'
 import {MapSvgByOblast} from '@/shared/maps/MapSvgByOblast'
+import {usePdmFilters} from '@/features/Meal/Pdm/Context/usePdmFilter'
 
-export interface DashboardPageProps {
-  filters: Record<string, string[]>
-  data: Seq<KoboAnswerFlat<Meal_cashPdm.T>>
+
+const isCashPdm = (_: PdmData<PdmForm>): _ is PdmData<Meal_cashPdm.T> => {
+  return _.type === 'Cash'
 }
 
 const mapOblast = OblastIndex.koboOblastIndexIso
 
-export const MealPdmDashboard = () => {
+export const MealPdmCashDashboard = () => {
   const ctx = useMealPdmContext()
   const ctxSchema = useKoboSchemaContext()
   const schema = ctxSchema.byName.meal_cashPdm.get!
   const langIndex = ctxSchema.langIndex
   const {m, formatDateTime, formatDate} = useI18n()
+  const {shape: commonShape, filters, setFilters} = usePdmFilters(ctx.fetcherAnswers.get)
   const [optionFilter, setOptionFilters] = useState<Record<string, string[] | undefined>>({})
-
   const filterShape = useMemo(() => {
-    return DataFilter.makeShape<KoboAnswerFlat<Meal_cashPdm.T>>({
-      oblast: {
-        icon: 'location_on',
-        getOptions: () => schema.schemaHelper.getOptionsByQuestionName('ben_det_oblast').map(_ => ({value: _.name, label: _.label[langIndex]})),
-        label: m.oblast,
-        getValue: _ => _.ben_det_oblast,
-      },
-      office: {
-        icon: 'share',
-        getOptions: () => schema.schemaHelper.getOptionsByQuestionName('office').map(_ => ({value: _.name, label: _.label[langIndex]})),
-        label: m.office,
-        getValue: _ => _.office,
-      },
-      projects: {
-        icon: appConfig.icons.sector,
-        getOptions: () => schema.schemaHelper.getOptionsByQuestionName('donor').map(_ => ({value: _.name, label: _.label[langIndex]})),
-        label: m.donor,
-        getValue: _ => _.donor
-      },
+    return DataFilter.makeShape({
```

Comment:
Here you need to explicitly define the type. Otherwise if `PdmData` or `Meal_cashPdm.T` change, compiler won't complain and the app will crashing on runtime.
```
DataFilter.makeShape<PdmData<Meal_cashPdm.T>>({
```